server/core/src/irods_signal.cpp

```diff
@@ -12,9 +12,12 @@
 #include <ctime>
 #include <system_error>
 #include <charconv>
+#include <string>
 
 #include <unistd.h>
 
+static const std::string STACKTRACE_DIR(irods::get_irods_stacktrace_directory().string());
```

Comment:
We need to think about when this line is initialized. It must be initialized on program startup to avoid violating async-signal safety.

Can the value returned by `irods::get_irods_stacktrace_directory()` be available at compile-time?

Reply:
No, for a couple of different reasons.
1. `irods::get_irods_stacktrace_directory()` returns a `boost::filesystem::path`. Since these objects are mutable, a new one must be constructed each time.
2. The values returned by `irods::get_irods_stacktrace_directory()` and friends are dependent on where `libirods_common.so.4.3.0` is located at runtime. This is required for relocatable installations.

Reply:
I've made `STACKTRACE_DIR` non-static and moved it into the `irods` namespace. I've also added a little logging to `irods::set_unrecoverable_signal_handlers()` that references the string so it is certain to be initialized before `stacktrace_signal_handler` is ever called.